frame/society/src/lib.rs

```diff
@@ -823,8 +829,36 @@ decl_module! {
 			Self::add_member(&founder)?;
 			<Head<T, I>>::put(&founder);
 			<Founder<T, I>>::put(&founder);
+			Rules::<T, I>::put(T::Hashing::hash(&rules));
 			Self::deposit_event(RawEvent::Founded(founder));
 		}
+
+		/// Anull the founding of the society.
+		///
+		/// The dispatch origin for this call must be Signed, and the signing account must be both
+		/// the `Founder` and the `Head`. This implies that it may only be done when there is one
```

Comment:
If I understand the code correctly, the Head will only be updated after some period. Couldn't this lead to a state at which members are already added, but the founder is still the head?

Reply:
candidates might be selected, but Head changes only with the content of Members afaik.

Reply:
Yeah there seems to be some problems with this:

Once one or more additional members are added to society, someone else is guaranteed to become the Head. Once they are Head, there is no way for them to be removed from society. So the Founder can never become Head once there is a second member. There will always be two members.

There is also an issue that @bkchr describes. At genesis, if you set initial members for the society, Founder will also be set as the Head, and with non-zero members, they would be able to call this function. Additionally, if somehow the founder did become both head and founder again as the last member (with some updated logic), it is possible that a suspended member becomes unsuspended, and then you will have non-zero members which satisfy these conditions.

Reply:
> with some updated logic

future changes that break current assumptions are hardly a concern here.